### ISSUE TITLE
Compilation of rust-ssl while mariadb is installed doesn't work

### ISSUE THREAD

**Issue by thakyZ** [sfackler/rust-openssl#1433]
While I attempt to build `rust-openssl` on `Ubuntu 20.04` it throws an error that `opensslconf.h` doesn't exist.   
I figured out that I needed to install a different OpenSSL version via #603. But when I went to `aptitude` to figure out what was preventing the specific version of OpenSSL to be installed.

And it turns out MariaDB from the deb-src `http://mariadb.mirror.globo.tech/repo/10.5/ubuntu` has a different SSL version it depends on. Now the SSL version has a conf.h, but it doesn't seem to want to compile against it when I copy it over to a new file with the right name.

So is there any way around this or do I have to just compile it after I temporarily uninstall MariaDB which isn't ideal.

**Comment by sfackler** [comment]
`conf.h` is a separate, unrelated bit of the OpenSSL API than `opensslconf.h`. What specific version of OpenSSL is that repository trying to install? As far as I'm aware, `opensslconf.h` has been part of OpenSSL's public API ~forever.

**Comment by thakyZ** [comment]
I figured it out apparently I was using a different repository than what was officially supported. If anyone encounters this issue it is the repo on the main MariaDB page you will want to use [here](https://downloads.mariadb.org/mariadb/repositories/) specifically Jalceo. Which is what worked for me. Also, @sfackler I am not sure if I recall what I tried because I have since gotten rid of the broken version.
